### ISSUE TITLE
False positive using ancestor pseudo-class (`vue-scoped-css/require-selector-used-inside`)

### ISSUE THREAD

**Issue by GrantGryczan** [future-architect/eslint-plugin-vue-scoped-css#433]
Minimal reproduction: https://play.vuejs.org/#eNp9kTFPwzAQhf+KdRNIVToglhAqAaoEDICA0UvkHG2KY1u+Swiq8t85u2rpUHWz3/ve6Z29hbsQiqFHKKFi7IKtGRfaKVU17aCMrYluNTASa1g8orW+mosjSDU/4uVK/GtRkfEBG1HKdU0XQl6qIqXVNg013vpYqojNjXZTmpFTgsMMWLLuq10VG/JO+uSEBuO70FqMr4Fb70hDuZuVvFr6/DxnjWOPs71u1mi+T+gbGpOm4S0iYRxQw8HjOq5Qtkz28uMFRzkfzM43vRX6jPmO5G2fOu6w+941UvuIy22fuuAjt271ScuR0dF+qVQ0kVPmNcinPJxZ/b/uVXGdc/KiMP0B8reZ3Q==

This results in the following nonsensical ESLint output, highlighting the space between `:has(div)` and `.test`:

```
  6:10  warning  The selector ` ` is unused in the template    vue-scoped-css/require-selector-used-inside
```

**Comment by GrantGryczan** [comment]
It isn't missing information.

**Comment by GrantGryczan** [comment]
Thanks!
